src/bundling/fuse.ts

```diff
@@ -1,137 +1,57 @@
-import { BindingsRewrite } from './bindings-rewrite'
-import { createHash } from 'crypto'
-import { readFileSync } from 'fs'
-import { dirname, join, basename } from 'path'
-
-function hashName(name: string | Buffer) {
-  return createHash('md5').update(name).digest('hex').toString().slice(0, 8)
-}
-
-export interface FuseBoxFile {
-  info: { absPath: string }
-  absPath: string
-  contents: string
-  analysis: {
-    dependencies: string[]
-    requiresRegeneration: boolean
+import { NexeCompiler } from '../compiler'
+import { FuseBox, JSONPlugin, CSSPlugin, HTMLPlugin, QuantumPlugin } from 'fuse-box'
+import { readFileAsync } from '../util'
+//import NativeModulePlugin from './fuse-native-module-plugin'
+
+function createBundle(filename: string, options: { name: string; minify: any; cwd: string }) {
+  const plugins: any = [JSONPlugin(), CSSPlugin(), HTMLPlugin()]
+  if (options.minify) {
+    plugins.push(
+      QuantumPlugin({
+        target: 'server',
+        uglify: true,
+        bakeApiIntoBundle: options.name
+      })
+    )
   }
-  loadContents(): void
-  makeAnalysis(
-    parsingOptions?: any,
-    traversalPlugin?: {
-      plugins: Array<{
-        onNode(file: FuseBoxFile, node: any, parent: any): void
-        onEnd(file: FuseBoxFile): void
-      }>
-    }
-  ): void
-}
-
-export interface NativeModulePluginOptions {
-  [key: string]:
-    | {
-        additionalFiles: string[]
-      }
-    | true
-}
-
-export default function(options: NativeModulePluginOptions) {
-  return new NativeModulePlugin(options)
+  const fuse = FuseBox.init({
+    cache: false,
+    log: Boolean(process.env.NEXE_BUNDLE_DEBUG) || false,
+    homeDir: options.cwd,
+    sourceMaps: false,
+    writeBundles: false,
+    output: '$name.js',
+    target: 'server',
+    plugins
+  })
+  fuse.bundle(options.name).instructions(`> ${filename}`)
+  return fuse.run().then(x => {
+    let output = ''
+    x.bundles.forEach(y => (output = y.context.output.lastPrimaryOutput.content!.toString()))
```

Comment:
since there is only 1 bundle, y not `x.bundles[0]...`?

Reply:
Its a Map

Reply:
lame